src/ctx.rs

```diff
@@ -99,6 +99,32 @@ impl<const UP_TO: usize, const WITH_DEFAULT: bool> CaseSetter<UP_TO, WITH_DEFAUL
         let mut buf = buf.index_mut(self.offset..self.offset + self.len);
         small_memset::<V, UP_TO, WITH_DEFAULT>(&mut *buf, val);
     }
+
+    /// # Safety
+    /// 
+    /// `buf` must be correctly aligned and dereferencable (but need not be
+    /// initialized).
+    #[inline]
+    pub unsafe fn set_raw<T: Clone + Copy>(&self, buf: *mut [T], val: T) {
+        assert!(buf.len() >= self.offset + self.len);
+        let buf = (buf as *mut T).add(self.offset);
+        match self.len {
+            01 if UP_TO >= 01 => unsafe { *(buf as *mut [T; 01]) = [val; 01] },
+            02 if UP_TO >= 02 => unsafe { *(buf as *mut [T; 02]) = [val; 02] },
+            04 if UP_TO >= 04 => unsafe { *(buf as *mut [T; 04]) = [val; 04] },
+            08 if UP_TO >= 08 => unsafe { *(buf as *mut [T; 08]) = [val; 08] },
+            16 if UP_TO >= 16 => unsafe { *(buf as *mut [T; 16]) = [val; 16] },
+            32 if UP_TO >= 32 => unsafe { *(buf as *mut [T; 32]) = [val; 32] },
+            64 if UP_TO >= 64 => unsafe { *(buf as *mut [T; 64]) = [val; 64] },
```

Comment:
These should use `.write`, too, since it's uninitialized to start with.

Reply:
I tried to use `.write` and I think it optimized differently. We don't need to use write here because we are writing literals with no drop impl (`T: Copy`).

Reply:
Ahh, that's annoying that it optimizes worse.  I'd mention that we're relying on `T: Copy` meaning there is no `impl Drop` here.

Reply:
I'll double check this tomorrow before landing.

Reply:
It doesn't seem to actually make a difference, so I went back to `ptr::write`